File: Pyvvo.Logistics.Model/Model/Warehouse.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Pyvvo.Logistics.Model
{
    public class Warehouse
    {
        [Required, Key] public long Id { get; set; }
        [Required] public long AddressId { get; set; }
        [Required] public long ContactId { get; set; }
        [Required] public long CreatedById { get; set; }
        [Required] public DateTime CreatedOn { get; set; }
        public DateTime UpdatedOn { get; set; }
        public bool IsActive { get; set; }
        public bool IsExternal { get; set; } 
        public Address Address { get; set; }
        public Contact Contact { get; set; }
        public User CreatedBy { get; set; }
        public List<User> Users { get; set; }
        public List<Note> Notes { get; set; }


    }
}

File: Pyvvo.Logistics/ClientApp/src/app/service/Model/warehouse.ts
import { Address } from "./Address";
import { User } from "./user";
import { Contact } from "./contact";
import { Note } from "./Note";

export class Warehouse {
    id: number;
    createdOn: Date;
    createdOnLocalTime: string;
    updatedOn: Date;
    name: string;
    isexternal: boolean;
    isactive: boolean;
    notes: Note[];
    address: Address;
    contact: Contact;
    createdBy: User;
}
